examples/3dsphere/deformation_flow.jl

```diff
@@ -0,0 +1,262 @@
+        q3 = 1.0
+    else
+        q3 = 0.1
+    end
+
+    if (z > z_c) && (ϕ > ϕ_c - rad2deg(1 / 8)) && (ϕ < ϕ_c + rad2deg(1 / 8))
+        q3 = 0.1
+    end
+    q4 = 1 - 3 / 10 * (q1 + q2 + q3)
+
+    ρq1 = ρ(z) * q1
+    ρq2 = ρ(z) * q2
+    ρq3 = ρ(z) * q3
+    ρq4 = ρ(z) * q4
+    ρq4 = 1.0
+
+    return (ρq1 = ρq1, ρq2 = ρq2, ρq3 = ρq3, ρq4 = ρq4)
+end
+
+function rhs!(dydt, y, (coords, face_coords), t)
+
+    ϕ = coords.lat
+    λ = coords.long
+    zc = coords.z
+    zf = face_coords.z
+    λp = λ .- 360 * t / τ
+    k = 10 * R / τ
+
+    ϕf = face_coords.lat
+    λf = face_coords.long
+    λpf = λf .- 360 * t / τ
+
+    sp =
+        @. 1 + exp((p_top - p_0) / b / p_top) - exp((p(zf) - p_0) / b / p_top) -
+           exp((p_top - p(zf)) / b / p_top)
+    ua = @. k * sind(λp)^2 * sind(2 * ϕ) * cos(pi * t / τ) +
+       2 * pi * R / τ * cosd(ϕ)
+    ud = @. ω_0 * R / b / p_top *
+       cosd(λp) *
+       cosd(ϕ)^2 *
+       cos(2 * pi * t / τ) *
+       (-exp((p(zc) - p_0) / b / p_top) + exp((p_top - p(zc)) / b / p_top))
+    uu = @. ua + ud
+    uv = @. k * sind(2 * λp) * cosd(ϕ) * cos(pi * t / τ)
+    ω = @. ω_0 * sind(λpf) * cosd(ϕf) * cos(2 * pi * t / τ) * sp
+    uw = @. -ω / ρ(zf) / grav
+
+    uₕ = Geometry.Covariant12Vector.(Geometry.UVVector.(uu, uv))
+    w = Geometry.Covariant3Vector.(Geometry.WVector.(uw))
+
+    ρq1 = y.ρq1
+    ρq2 = y.ρq2
+    ρq3 = y.ρq3
+    ρq4 = y.ρq4
+
+    dρq1 = dydt.ρq1
+    dρq2 = dydt.ρq2
+    dρq3 = dydt.ρq3
+    dρq4 = dydt.ρq4
+
+    If2c = Operators.InterpolateF2C()
+    Ic2f = Operators.InterpolateC2F(
+        bottom = Operators.Extrapolate(),
+        top = Operators.Extrapolate(),
+    )
+    vdivf2c = Operators.DivergenceF2C(
+        top = Operators.SetValue(Geometry.Contravariant3Vector(0.0)),
+        bottom = Operators.SetValue(Geometry.Contravariant3Vector(0.0)),
+    )
+    hdiv = Operators.Divergence()
+
+    cw = If2c.(w)
+    cuvw = Geometry.Covariant123Vector.(uₕ) .+ Geometry.Covariant123Vector.(cw)
+
+    @. dρq1 = -hdiv(cuvw * ρq1)
+    @. dρq1 -= vdivf2c(w * Ic2f(ρq1))
+    @. dρq1 -= vdivf2c(Ic2f(uₕ * ρq1))
+
+    @. dρq2 = -hdiv(cuvw * ρq2)
+    @. dρq2 -= vdivf2c(w * Ic2f(ρq2))
+    @. dρq2 -= vdivf2c(Ic2f(uₕ * ρq2))
+
+    @. dρq3 = -hdiv(cuvw * ρq3)
+    @. dρq3 -= vdivf2c(w * Ic2f(ρq3))
+    @. dρq3 -= vdivf2c(Ic2f(uₕ * ρq3))
+
+    @. dρq4 = -hdiv(cuvw * ρq4)
+    @. dρq4 -= vdivf2c(w * Ic2f(ρq4))
+    @. dρq4 -= vdivf2c(Ic2f(uₕ * ρq4))
+
+    Spaces.weighted_dss!(dydt)
+
+    return dydt
+end
+
+dydt = similar(y0)
+rhs!(dydt, y0, (coords, face_coords), 0.0)
+
+# run!
+T = 86400 * 12
+dt = 60 * 60
+prob = ODEProblem(rhs!, y0, (0.0, T), (coords, face_coords))
+sol = solve(
+    prob,
+    SSPRK33(),
+    dt = dt,
+    saveat = dt,
+    progress = true,
+    adaptive = false,
+    progress_message = (dt, u, p, t) -> t,
+)
+
+q1_error =
+    norm(sol.u[end].ρq1 ./ ρ.(coords.z) .- y0.ρq1 ./ ρ.(coords.z)) /
+    norm(sol.u[end].ρq1 ./ ρ.(coords.z))
```

Comment:
how about dividing the differences by the initial states?

Reply:
same for the other tracers

Reply:
done:)